modules/investigating-worker-node-installation-issues.adoc

```diff
@@ -0,0 +1,200 @@
+====
++
+. Retrieve `crio.service` journald unit logs on worker nodes, after they have booted. This provides visibility into worker node CRI-O container runtime activity.
+.. Retrieve the logs using `oc`:
++
+----
+$ oc adm node-logs --role=worker -u crio
+----
++
+.. If the API is not functional, review the logs using SSH instead:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> journalctl -b -f -u crio.service
+----
+
+. Collect logs from specific subdirectories under `/var/log/` on worker nodes.
+.. Retrieve a list of logs contained within a `/var/log/` subdirectory. The following example lists files in `/var/log/sssd/` on all worker nodes:
++
+----
+$ oc adm node-logs --role=worker --path=sssd
+----
++
+.. Inspect a specific log within a `/var/log/` subdirectory. The following example outputs `/var/log/sssd/audit.log` contents from all worker nodes:
++
+----
+$ oc adm node-logs --role=worker --path=sssd/sssd.log
+----
++
+.. If the API is not functional, review the logs on each node using SSH instead. The following example tails `/var/log/sssd/sssd.log`:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> sudo tail -f /var/log/sssd/sssd.log
+----
+
+. Review worker node container logs using SSH.
+.. List the containers:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> sudo crictl ps -a
+----
++
+.. Retrieve a container's logs using `crictl`:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> sudo crictl logs -f <container_id>
+----
+
+. If you experience worker node configuration issues, verify that the MCO, MCO endpoint and DNS record are functioning. The Machine Config Operator (MCO) manages operating system configuration during the installation procedure. Also verify system clock accuracy and certificate validity.
+.. Test whether the MCO endpoint is available. Replace `<cluster_name>` with appropriate values:
++
+----
+$ curl https://api-int.<cluster_name>:22623/config/worker
+----
++
+.. If the endpoint is unresponsive, verify load balancer configuration. Ensure that the endpoint is configured to run on port 22623.
++
+.. Verify that the MCO endpoint's DNS record is configured and resolves to the load balancer.
+... Run a DNS lookup for the defined MCO endpoint name:
++
+----
+$ dig api-int.<cluster_name> @<dns_server>
+----
++
+... Run a reverse lookup to the assigned MCO IP address on the load balancer:
++
+----
+$ dig -x <load_balancer_mco_ip_address> @<dns_server>
+----
++
+.. Verify that the MCO is functioning from the bootstrap node directly. Replace `<bootstrap_fqdn>` with bootstrap node's fully qualified domain name:
```

Comment:
s/with bootstrap node’s fully/with the bootstrap node’s fully/
`grep` for a few other instances of this as well

Reply:
Resolved throughout the PR.